tests/SprykerEcoTest/Zed/Braintree/Business/BraintreeFacadeCheckoutPostSaveHookTest.php

```diff
@@ -0,0 +1,180 @@
+     */
+    public function testCheckoutPostSaveHookWithSuccessfulResponse()
+    {
+        $response = $this->executeCheckoutPostSaveHook();
+        $this->assertTrue($response->getIsSuccess());
+    }
+
+    /**
+     * @return void
+     */
+    public function testCheckoutPostSaveHookWithErrorResponse()
+    {
+        $response = $this->executeCheckoutPostSaveHook(false);
+        $this->assertFalse($response->getIsSuccess());
+    }
+
+    /**
+     * @param bool $isSuccess
+     *
+     * @return @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    protected function executeCheckoutPostSaveHook(bool $isSuccess = true): CheckoutResponseTransfer
+    {
+        $factoryMock = $this->getFactoryMock(['createPaymentTransaction', 'createOrderSaver']);
+        $factoryMock->expects($this->once())->method('createPaymentTransaction')->willReturn(
+            $this->getPaymentTransactionMock($isSuccess)
+        );
+        $factoryMock->expects($this->once())->method('createOrderSaver')->willReturn(
+            $this->getOrderSaverMock()
+        );
+        $braintreeFacade = $this->getBraintreeFacade($factoryMock);
+
+        $orderTransfer = $this->createOrderTransfer();
+        $quoteTransfer = $this->getQuoteTransfer($orderTransfer);
+        $checkoutResponseTransfer = $this->getCheckoutResponseTransfer($orderTransfer);
+
+        return $braintreeFacade->executeCheckoutPostSaveHook($quoteTransfer, $checkoutResponseTransfer);
+    }
+
+    /**
+     * @param bool $success
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function getPaymentTransactionMock(bool $success = true)
+    {
+        $moneyFacadeMock = $this->getMoneyFacadeMock();
+        $paymentTransactionMock = $this
+            ->getMockBuilder(PaymentTransaction::class)
+            ->setMethods(['doTransaction', 'initializeBraintree'])
+            ->setConstructorArgs(
+                [new BraintreeConfig(), new BraintreeToMoneyFacadeBridge($moneyFacadeMock)]
+            )
+            ->getMock();
+
+        $doRequestResponse = $success ? $this->getSuccessResponse() : $this->getErrorResponse();
+        $paymentTransactionMock->expects($this->once())->method('doTransaction')->willReturn($doRequestResponse);
+
+        return $paymentTransactionMock;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function getOrderSaverMock()
+    {
+        return $this
+            ->getMockBuilder(Saver::class)
+            ->setConstructorArgs([new BraintreeEntityManager()])
+            ->getMock();
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject
+     */
+    protected function getMoneyFacadeMock()
+    {
+        return $this
+            ->getMockBuilder(BraintreeToMoneyFacadeInterface::class)
+            ->getMock();
+    }
+
+    /**
+     * @return \Braintree\Result\Successful
+     */
+    protected function getSuccessResponse()
```

Comment:
Please move to tester anything that is not a test, and not a `mock` creation method.

Reply:
Updated